src/rdkafka_conf.c

```diff
@@ -1011,19 +1011,20 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
 
         /* Topic producer properties */
         { _RK_TOPIC|_RK_PRODUCER|_RK_HIGH, "request.required.acks", _RK_C_INT,
-	  _RKT(required_acks),
-	  "This field indicates how many acknowledgements the leader broker "
-	  "must receive from ISR brokers before responding to the request: "
-	  "*0*=Broker does not send any response/ack to client, "
-	  "*1*=Only the leader broker will need to ack the message, "
-	  "*-1* or *all*=broker will block until message is committed by all "
-	  "in sync replicas (ISRs) or broker's `min.insync.replicas` "
-	  "setting before sending response. ",
-	  -1, 1000, 1,
-	  .s2i = {
-			{ -1, "all" },
-		}
-	},
+          _RKT(required_acks),
+          "This field indicates the number of acknowledgements the leader "
+          "broker must receive from ISR brokers before responding to the "
```

Comment:
>ISR brokers

The broker isn't really a member of the ISR the replica is. This is an important distinction because one healthy partition does not necessarily mean you have an unhealthy broker.

Reply:
"This is an important distinction because one healthy partition does not necessarily mean you have an unhealthy broker."
Not really following that

Reply:
All brokers: 1,2,3,4,5 
All replicas: 1,2,3,4,5
Leader 2 ISR list for TopicPartition `tp-1`: [1,2,4,5]

This does not necessarily mean that all replicas held by broker 3 are out of sync. It means that replica 3 for Topic-Partition `tp-1` is out of sync. Calling the ISR member `ISR Brokers` implies that the broker has fallen out of sync rather than the replica that lives on that broker.

Reply:
According to TopicMetadata the leader considers itself part of the ISR:
```
  topic "test" with 4 partitions:
    partition 0, leader 4, replicas: 4,2,3, isrs: 4,2,3
```

Reply:
As it should be but again this is reporting the state of the replica not necessarily the broker.

Reply:
From a producer perspective which thinks only at the TopicPartition level  I guess this is not true.